### ISSUE TITLE
Undo broken in Ferdi 5.4.1-beta.4

### ISSUE THREAD

**Issue by ylluminate** [getferdi/ferdi#296]
Unfortunately cmd-z (also edit -> undo) in macOS (Catalina; 10.15.2) has stopped working within Mattermost in Ferdi 5.4.1-beta.4.  Undo still seems to work in other services such as Discord.

**Comment by ylluminate** [comment]
Anyone have any thoughts on this? This is proving extremely difficult to work with and I'm finding myself having to go back to Rambox in order to use Mattermost.

**Comment by ylluminate** [comment]
Just checking in to verify that this is still broken in 5.4.3.

**Comment by ylluminate** [comment]
Can I provide any additional info for this?

**Comment by kytwb** [comment]
@ylluminate Sorry for the inconvenience. Someone will have to try and reproduce before being able to debug; do you have any error that could give some hints in the consoles? (View > Developer Tools && Service Developer Tools)

**Comment by ylluminate** [comment]
Hey sure thing - when I fire up those consoles I'm not seeing anything actively generated on cmd-z. Let me poke around a bit here...

**Comment by ylluminate** [comment]
So yeah, unfortunately when I issue `cmd-z` I don't see any errors triggered in either one, only the ones noted in the logs above. Is there a way to look further into this regarding what is being triggered when I press those keys within Ferdi?

**Comment by kytwb** [comment]
Could it be related to https://mattermost.atlassian.net/browse/MM-19198 and https://github.com/mattermost/desktop/pull/1092/files?

**Comment by ylluminate** [comment]
Well, I'd like to believe that could be the case, but unfortunately it doesn't seem so. I can use both the desktop release of their app  (4.3.2 (4.3.2.4230)) and Rambox Pro with success here. Ferdi is the exception.

**Comment by kytwb** [comment]
Not sure if related but we had other reports related to shortcuts not working via https://github.com/getferdi/ferdi/issues/362 and https://github.com/getferdi/ferdi/issues/377.

**Comment by aleksabobic** [comment]
Hey @kytwb and @ylluminate  just wanted to clarify my bug report ( #377 ) was not related to mattermost but to messenger / telegram / whatsapp so this might be a global issue on Macs.

**Comment by aleksabobic** [comment]
After investigation I found the issue is most likely caused by: https://github.com/getferdi/ferdi/commit/9214b63967ca659fb083d9c994842995aef126f6
There are no menu items hence there is no possibility of using the shortcuts. Unfortunatelly I don't have time right now to fix this but it would be great if someone could either just provide the "Edit" sub-menu even when Ferdi is locked or cherry pick all the menu items which don't enable the user to unlock Ferdi. 

Additionally the locked flag is set to true by default when you start Ferdi. Should this be expected behavior.

**Comment by kytwb** [comment]
@aleksabobic 

> There are no menu items hence there is no possibility of using the shortcuts.

Good point! I was wondering why I couldn't CTRL+A to erase the lock password when making a typo! 🎉  I will add a default menu when locked so we have access to default shortcuts.  
On the other hand, I don't understand how that could affect shortcuts when not locked; I tried to see if shortcuts where affected after locking/unlocking, and they are not (still working for me, on macOS).

> Additionally the locked flag is set to true by default when you start Ferdi

If you enable Ferdi Lock, it starts locked, yes.

**Comment by kytwb** [comment]
@aleksabobic is your menu empty when your shortcuts don't work?

**Comment by aleksabobic** [comment]
Yes exactly. When the menu is empty then the shortcuts aren't working.

**Comment by kytwb** [comment]
I got it 👌 It indeed has to do with Ferdi Lock. Looking into it.

**Comment by kytwb** [comment]
@aleksabobic I found the issue, it's in Ferdi Lock:

<img width="813" alt="Screen Shot 2020-02-22 at 2 09 34 AM" src="https://user-images.githubusercontent.com/412895/75083330-7775b400-5518-11ea-9f18-969e68df42c5.png">

I will make a pull request asap and update #362.
Let's keep this issue to keep track of @ylluminate issue with Mattermost only.

**Comment by ylluminate** [comment]
Still broken here in 5.4.4-beta.2.

**Comment by ylluminate** [comment]
Still broken in 5.5.0-gm (5.5.0-gm.1893).

**Comment by kytwb** [comment]
@ylluminate Thanks for your patience! 🙏 
We are closing on long standing issues and will soon enough be able to address other pending ones! 💪
